include/zephyr/drivers/i3c.h

```diff
@@ -1111,24 +1320,167 @@ static inline int i3c_recover_bus(const struct device *dev)
  * been updated with SETNEWDA.
  * This method is optional.
  *
+ * @warning
+ * Use cases involving multiple writers to the i3c/i2c devices must prevent
+ * concurrent write operations, either by preventing all writers from
+ * being preempted or by using a mutex to govern writes to the i3c/i2c devices.
+ *
  * @param target Pointer to the target device descriptor
  * @param old_dyn_addr The old dynamic address of target device, 0 if
  *            there was no old dynamic address
  *
  * @retval 0 If successful.
  * @retval -EINVAL If address is not available
  */
-static inline int i3c_reattach_i3c_device(struct i3c_device_desc *target,
-			uint8_t old_dyn_addr)
+static inline int i3c_reattach_i3c_device(struct i3c_device_desc *target, uint8_t old_dyn_addr)
 {
-	const struct i3c_driver_api *api =
-		(const struct i3c_driver_api *)target->bus->api;
+	struct i3c_driver_data *data = (struct i3c_driver_data *)target->bus->data;
+	const struct i3c_driver_api *api = (const struct i3c_driver_api *)target->bus->api;
+	int status = 0;
```

Comment:
That's not the intent. `api->reattach_i3c_device` is an optional api to adjust retaining registers (if the IP has it). Returning `-ENOSYS` could be confusing if using the Silvca (NXP) IP which does not require any of the retaining register updates. Essentially all this function would do is update the address slots if that address is free.

Reply:
The device API usually calls for `-ENOSYS` if the API is not implemented at the driver level. And there is (almost) always 1-to-1 relations between this type of stubs and corresponding APIs. Probably a good idea to mention this in the API doc if this still works without driver implementation.

Reply:
I though `-ENOSYS` would be if the _"entire"_ function is not available which isn't exactly the case here. Id rather see 0 for the case of portability between switching between different IP if it is completely successful. I can add an extra note to the api doxygen about how it will call an optional api if implemented.